pkg/splunk/client/minioclient.go

```diff
@@ -73,8 +73,8 @@ func NewMinioClient(ctx context.Context, bucketName string, accessKeyID string,
 
 //RegisterMinioClient will add the corresponding function pointer to the map
 func RegisterMinioClient() {
-	wrapperObject := GetS3ClientWrapper{GetS3Client: NewMinioClient, GetInitFunc: InitMinioClientWrapper}
-	S3Clients["minio"] = wrapperObject
+	wrapperObject := GetRemoteDataClientWrapper{GetRemoteDataClient: NewMinioClient, GetInitFunc: InitMinioClientWrapper}
+	RemoteDataClientsMap["minio"] = wrapperObject
```

Comment:
I have my doubts in using map here. I do not think we ever plan to support multiple cloud services at the same time ? . are we over designing here ?

Reply:
Per my discussion with some of the folks (@gaurav-splunk / @sgontla ), plan is to keep support for multiple cloud services. it looks a little vague but if one cloud provider (e.g aks cluster) has access to another cloud provider (like AWS S3) then our design should work. More to be covered in testing.

Reply:
We do support multiple cloud providers for different volumes for the same CRD, hence this.